### DISCUSSION TITLE
Proposal: Set the default branch to Main instead of Develop

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#1665] [Ideas] [answered]
Here is a handy use case that could really benefit from main being the default branch:  

For a simpler developer experience when creating oqtane modules, we add the oqtane.framework as a sub-module to our repo and then add a few oqtane projects to our solution.  This allows us to use a single instance of Visual Studio to debug our modules.

The issue we are running into with develop being the default branch is that there are a few extra steps to set the submodule to the proper version.  It would be extra handy to have the default branch to be default for this case.

I understand that having develop be the default is handy for other reasons.

**Comment by leigh-pointer** [comment]
To be honest you only need to a the oqtane server project to the external solution and make sure this is the start project.
I have been doing this since i started developing for the oqtane framework.
________________________________
From: Mark Davis ***@***.***>
Sent: Friday, September 17, 2021 2:25:28 AM
To: oqtane/oqtane.framework ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [oqtane/oqtane.framework] Proposal: Set the default branch to Main instead of Develop (Discussion #1665)


Here is a handy use case that could really benefit from main being the default branch:

For a simpler developer experience when creating oqtane modules, we add the oqtane.framework as a sub-module to our repo and then add a few oqtane projects to our solution. This allows us to use a single instance of Visual Studio to debug our modules.

The issue we are running into with develop being the default branch is that there are a few extra steps to set the submodule to the proper version. It would be extra handy to have the default branch to be default for this case.

I understand that having develop be the default is handy for other reasons.

**Reply by markdav-is** [reply]
indeed.  before we were using sub-modules, we just had instructions in the readme to get rolling with these steps.  The sub-module reduces a few steps in the setup process.  unfortunately, it also adds one step that is to change the branch in the submodule.

**Comment by markdav-is** [comment, marked as answer]
I'm going to answer my own question here .  This is not required because you can specify the branch in you gitmodules file

[submodule "src/oqtane.framework"]
	path = src/oqtane.framework
	url = https://github.com/oqtane/oqtane.framework
	branch = master

more info: https://www.git-scm.com/docs/gitmodules
